python-mini-projects/projects/Get_meta_information_of_images/get_meta_from_pic.py

```diff
@@ -0,0 +1,60 @@
+from PIL import Image
+from PIL.ExifTags import TAGS
+from author_utils import get_file_security, get_author
+from gps_utils import get_location
+import os
+import sys
+from datetime import datetime
+
+def get_exif(image):
+    image.verify()
+    return image._getexif()
+
+
+def get_labeled_exif(exif):
+    labeled = {}
+    for (key, val) in exif.items():
+        labeled[TAGS.get(key)] = val
+
+    return labeled
+
+im = Image.open(sys.argv[1])
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *im*. The resource is allocated by call *Image.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *im* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).